command/pki_issue_intermediate.go

```diff
@@ -0,0 +1,351 @@
+	resp, err := client.Logical().Read(sanitizePath(intermediateMount + "/issuer/" + issuerId))
+	if err != nil || resp == nil {
+		c.UI.Error(fmt.Sprintf("Error Reading Fully Imported Certificate from %v : %v",
+			intermediateMount+"issuer/"+issuerId, err))
+	}
+
+	OutputSecret(c.UI, resp)
+}
+
+func importIssuerWithName(client *api.Client, mount string, bundle string, name string) (issuerUUID string, err error) {
+	importData := map[string]interface{}{
+		"pem_bundle": bundle,
+	}
+	writeResp, err := client.Logical().Write(mount+"/issuers/import/cert", importData)
+	if err != nil {
+		return "", err
+	}
+	mapping := writeResp.Data["mapping"].(map[string]interface{})
+	if len(mapping) > 1 {
+		return "", fmt.Errorf("multiple issuers returned, while expected one, got %v", writeResp)
+	}
+	for issuerId := range mapping {
+		issuerUUID = issuerId
+	}
+	if name != "" && name != "default" {
+		nameReq := map[string]interface{}{
+			"issuer_name": name,
+		}
+		ctx := context.Background()
+		_, err = client.Logical().JSONMergePatch(ctx, mount+"/issuer/"+issuerUUID, nameReq)
+		if err != nil {
+			return issuerUUID, fmt.Errorf("error naming issuer %v to %v: %v", issuerUUID, name, err)
+		}
+	}
+	return issuerUUID, nil
+}
+
+type inCaseOfFailure struct {
+	csrGenerated    bool
+	csrSigned       bool
+	certImported    bool
+	certNamed       bool
+	caChainImported bool
+
+	intermediateMount string
+	createdKeyId      string
+	csr               string
+	caChain           string
+	parentMount       string
+	parentIssuer      string
+	certSerialNumber  string
+	certIssuerId      string
+	newName           string
+}
+
+func (state inCaseOfFailure) generateFailureMessage() string {
+	message := "A failure has occurred"
+
+	if state.csrGenerated {
+		message += fmt.Sprintf(" after \n a Certificate Signing Request was successfully generated on mount %v", state.intermediateMount)
+	}
+	if state.csrSigned {
+		message += fmt.Sprintf(" and after \n that Certificate Signing Request was successfully signed by mount %v", state.parentMount)
+	}
+	if state.certImported {
+		message += fmt.Sprintf(" and after \n the signed certificate was reimported into mount %v , with issuerID %v", state.intermediateMount, state.certIssuerId)
+	}
+
+	if state.csrGenerated {
+		message += "\n\nTO CONTINUE: \n" + state.toContinue()
+	}
+	if state.csrGenerated && !state.certImported {
+		message += "\n\nTO ABORT: \n" + state.toAbort()
+	}
+
+	message += "\n"
+
+	return message
+}
+
+func (state inCaseOfFailure) toContinue() string {
+	message := ""
+	if !state.csrSigned {
+		message += fmt.Sprintf("You can continue to work with this Certificate Signing Request CSR PEM, by saving"+
+			" it as `pki_int.csr`: %v \n Then call `vault write %v/sign-intermediate csr=@pki_int.csr ...` adding the "+
+			"same key-value arguements as to `pki issue` (except key_type and issuer_name) to generate the certificate "+
+			"and ca_chain", state.csr, state.parentIssuer)
```

Comment:
Should this not be `state.parentMount`?

Reply:
I don't think so.  state.parentIssuer is the (fully qualified) parentMountIssuer we take from the arguments.  

```	
	failureState := inCaseOfFailure{
		intermediateMount: intermediateMount,
		parentMount:       strings.Split(parentMountIssuer, "/issuer/")[0],
		parentIssuer:      parentMountIssuer,
		newName:           c.flagNewIssuerName,
	}
```

Reply:
Ah yup, yup I'm incorrect, sorry!